File: programmers/python 3/17687.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/17687

def solution(n, t, m, p):
    s = ''
    i = 0
    while len(s) < t * m:
        s += based(i, n)
        i += 1

    return ''.join(s[i] for i in range(p - 1, t * m, m))

def based(n, base):
    if not n:
        return '0'

    nums = '0123456789ABCDEF'
    result = ''
    while n:
        result += nums[n % base]
        n //= base

    return result[::-1]

File: programmers/swift/17687.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/17687

func solution(_ n: Int, _ t: Int, _ m: Int, _ p: Int) -> String {
    var s = [String]()
    var i = 0
    while s.count < t * m {
        s += String(i, radix: n).uppercased().map { String($0) }
        i += 1
    }

    return stride(from: p - 1, to: t * m, by: m)
        .map { s[$0] }
        .joined()
}
